shotover-proxy/src/protocols/redis_codec.rs

```diff
@@ -587,26 +544,30 @@ impl Decoder for RedisCodec {
     type Item = Messages;
     type Error = anyhow::Error;
 
-    fn decode(
-        &mut self,
-        src: &mut BytesMut,
-    ) -> std::result::Result<Option<Self::Item>, Self::Error> {
-        Ok(match self.decode_raw(src)? {
-            None => None,
-            Some(f) => Some(self.process_redis_bulk(f)?),
-        })
+    fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Self::Item>> {
+        let mut messages = vec![];
```

Comment:
doesn't this mean that we create a new vec and then allocate on each push in the loop. We do this for each new call to decode (which happens a lot)?

Reply:
This should have the same amount of allocations as the previous implementation.
* Empty vecs dont allocate. - Once we push a single item on an empty vec it allocates for 8 items.
* Every time we called `std::mem::take` in the previous implementation that would return the previous vec and leave a new empty vec in its place. - equivalent to just creating an empty vec every time.